cmd/p2p_snake/main.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/gdamore/tcell/v2"
+)
+
+func drawText(s tcell.Screen, x1, y1, x2, y2 int, style tcell.Style, text string) {
+	row := y1
+	col := x1
+	for _, r := range []rune(text) {
+		s.SetContent(col, row, r, nil, style)
+		col++
+		if col >= x2 {
+			row++
+			col = x1
+		}
+		if row > y2 {
+			break
+		}
+	}
+}
+
+func drawInitialBox(s tcell.Screen, boundary Boundary, style tcell.Style) {
+	x1, y1 := boundary.topLeft.x, boundary.topLeft.y
+	x2, y2 := boundary.bottomRight.x, boundary.bottomRight.y
+	if y2 < y1 {
+		y1, y2 = y2, y1
+	}
+	if x2 < x1 {
+		x1, x2 = x2, x1
+	}
+
+	// Fill background
+	for row := y1; row <= y2; row++ {
+		for col := x1; col <= x2; col++ {
+			s.SetContent(col, row, tcell.RuneBullet, nil, style)
+		}
+	}
+
+	// Draw borders
+	for col := x1; col <= x2; col++ {
+		s.SetContent(col, y1, tcell.RuneHLine, nil, style)
+		s.SetContent(col, y2, tcell.RuneHLine, nil, style)
+	}
+	for row := y1 + 1; row < y2; row++ {
+		s.SetContent(x1, row, tcell.RuneVLine, nil, style)
+		s.SetContent(x2, row, tcell.RuneVLine, nil, style)
+	}
+
+	// Only draw corners if necessary
+	if y1 != y2 && x1 != x2 {
+		s.SetContent(x1, y1, tcell.RuneULCorner, nil, style)
+		s.SetContent(x2, y1, tcell.RuneURCorner, nil, style)
+		s.SetContent(x1, y2, tcell.RuneLLCorner, nil, style)
+		s.SetContent(x2, y2, tcell.RuneLRCorner, nil, style)
+	}
+
+	//drawText(s, x1+1, y1+1, x2-1, y2-1, style, text)
+}
+
+type Coord struct {
+	x, y int
+}
+
+type Snake struct {
+	id    int         // snake id
+	body  []Coord     // coordinates of snake's body
+	head  Coord       // coordinates of snake's head
+	style tcell.Style // snake's style
+}
+
+type Boundary struct {
+	topLeft     Coord
+	bottomRight Coord
+}
+
+func drawSnake(s tcell.Screen, snake Snake, boundary Boundary) error {
+	if (snake.head.x <= boundary.topLeft.x || snake.head.x >= boundary.bottomRight.x) ||
+		(snake.head.y <= boundary.topLeft.y || snake.head.y >= boundary.bottomRight.y) {
```

Comment:
This piece of code is repeated more than twice. Move it to a dedicated function, like `Boundary.Contains(p Point) bool`, please.